gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -787,119 +789,191 @@ local function FindCorpse(ply)
+
+			if not spawnPos then
+				local spawnEntity = spawn.GetRandomPlayerSpawnEntity(ply)
+
+				spawnPos = spawnEntity:GetPos()
+				spawnEyeAngle = spawnEntity:EyeAngles()
+			end
 
+			ply:SetPos(spawnPos)
+			ply:SetEyeAngles(spawnEyeAngle or Angle(0, 0, 0))
 			ply:SetMaxHealth(100)
 
+			hook.Run("PlayerLoadout", ply)
+
 			local credits = CORPSE.GetCredits(corpse, 0)
 
 			ply:SetCredits(credits)
+			ply:SelectWeapon("weapon_zm_improvised")
 
 			if IsValid(corpse) then
 				corpse:Remove()
 			end
 
 			DamageLog("TTT2Revive: " .. ply:Nick() .. " has been respawned.")
 
-			if isfunction(fn) then
-				fn(ply)
+			if isfunction(OnRevive) then
+				OnRevive(ply)
 			end
-		elseif isfunction(onFail) then
-			ply:ChatPrint("Revive failed...")
-
-			onFail(ply)
+		else
+			OnReviveFailed(ply, "message_revival_failed")
 		end
 	end)
 end
 
 ---
--- Returns if a player is currently in a revival process started by @{Player:Revive}
--- @return boolean The revival status
+-- Cancel the ongoing revival process.
+-- @param [default="message_revival_canceled"]string failMessage The fail message that should be displayed for the client
+-- @realm server
+function plymeta:CancelRevival(failMessage)
+	self:SetReviving(false)
+	self:SetBlockingRevival(false)
+	self:SetRevivalReason(nil)
+
+	timer.Remove("TTT2RevivePlayer" .. self:EntIndex())
+
+	OnReviveFailed(self, failMessage or "message_revival_canceled")
+end
+
+---
+-- Sets the revival state.
+-- @param [default=false]boolean isReviving The reviving state
+-- @realm server
+function plymeta:SetReviving(isReviving)
+	self:TTT2NETSetBool("player_is_reviving", isReviving or false)
+end
+
+---
+-- Sets the blocking revival state
+-- @param [default=false]boolean isBlockingRevival The blocking revival state
+-- @realm server
+function plymeta:SetBlockingRevival(isBlockingRevival)
+	self:TTT2NETSetBool("player_is_blocking_revival", isBlockingRevival or false)
+end
+
+---
+-- Sets the revival start time
+-- @param [default=@{CurTime()}]number startTime The revival start time
+-- @realm server
+function plymeta:SetRevivalStartTime(startTime)
+	self:TTT2NETSetFloat("player_revival_start_time", startTime or CurTime())
+end
+
+---
+-- Sets the revival time
+-- @param [default=0.0]number time The revival time
+-- @realm server
+function plymeta:SetRevivalTime(time)
+	self:TTT2NETSetFloat("player_revival_time", time or 0.0)
+end
+
+---
+-- Sets the death position
+-- @param Vector pos The death position
 -- @realm server
-function plymeta:IsReviving()
-	return self.reviving or false
+function plymeta:SetDeathPosition(pos)
+	self:TTT2NETSetFloat("player_death_pos_x", pos.x or 0.0)
+	self:TTT2NETSetFloat("player_death_pos_y", pos.y or 0.0)
+	self:TTT2NETSetFloat("player_death_pos_z", pos.z or 0.0)
+end
+
+---
+-- Sets the spawn position
+-- @param Vector pos The spawn position
+-- @realm server
+function plymeta:SetSpawnPosition(pos)
+	self:TTT2NETSetFloat("player_spawn_pos_x", pos.x or 0.0)
```

Comment:
You could put them in their own path, to group the values: `"player_spawn_pos_x"` -> `{"player_spawn_pos", "x"}`

Reply:
Also the system is capable of setting values to `nil`, which should be considered instead of 0.0 as a default value

Reply:
Please also note, that this will result in all players knowing about the spawn position of each other player.